### PR TITLE
react implementation

### PR DESCRIPTION
**Pull request by Sanjit-Prasad**
Implemented 'Schedule' aka 'Interview' app in react.
Use api calls to fetch data and call methods of controller.
home
interviews
create
show

### CODE REVIEW

File: src/Bottombar.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+function Bottombar() {
+    return (
+      <div>
+        <section className="section">
+        <div id="footer_container">
+                <footer className="footer">
+                    <div className="content has-text-centered">
+                        <p>
+                            For any query, email at : sanjitpd777@gmail.com
```
Comment (joydeep-ib) [inline src/Bottombar.js:12]: Ohh

File: src/Bottombar.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+                        </p>
+                    </div>
+                </footer>
+            </div>        </section>
```
Comment (joydeep-ib) [inline src/Bottombar.js:16]: Indentation

File: src/Delete.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import './App.css'
+
+function Delete() {
+    const pageUrl = window.location.href
+    const id = pageUrl.substring(35,38);
+    
+    var xhttp = new XMLHttpRequest();
```
Comment (joydeep-ib) [inline src/Delete.js:7]: better use fetch api

File: src/Delete.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    xhttp.onreadystatechange = function() {
+      if (this.readyState === 4 && this.status === 200) {
+        document.getElementById("demo").innerHTML = "Interview successfully deleted.";
+      }
+    };
+    xhttp.open("DELETE", "http://localhost:3000/interviews/" + id, true);
+    xhttp.send();
+    
+    window.location.href = "/"
```
Comment (joydeep-ib) [inline src/Delete.js:16]: Not recommended. Use history instead

File: src/App.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import React from 'react'
+import { HashRouter as Router, Switch, Route } from  'react-router-dom';
```
Comment (joydeep-ib) [inline src/App.js:2]: Use browser-router & history

File: src/Delete.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import './App.css'
+
+function Delete() {
```
Comment (joydeep-ib) [inline src/Delete.js:3]: Delete is pretty generic & can be confused with `delete`. Try not to use very generic names

File: src/Edit.js [hunk 7]
```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import './App.css';
+import {Link} from  'react-router-dom';
+
+function EditInterview(event) {
```
Comment (joydeep-ib) [inline src/Edit.js:5]: Better named

File: src/Delete.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+    const pageUrl = window.location.href
+    const id = pageUrl.substring(35,38);
```
Comment (joydeep-ib) [inline src/Delete.js:5]: Not a very good way to access the id. What if the route changes? are you going to manually edit 35 & 38 ? What if we have interviews > 10,000? Your app is going to break. There is a hook for what you're trying to achieve.

File: src/Edit.js [hunk 9]
```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import './App.css';
+import {Link} from  'react-router-dom';
+
+function EditInterview(event) {
+  
+  const pageUrl = window.location.href
+  const id = pageUrl.substring(35,38);
```
Comment (joydeep-ib) [inline src/Edit.js:8]: Use hook here

File: src/Edit.js [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+    console.log(json_object);
+    
+    const options = 
+      {
+          method: 'PATCH',            
+          mode: 'cors',
+          headers: {
+              'Content-Type': 'application/json'
+          },
+          body: JSON.stringify(json_object),
```
Comment (joydeep-ib) [inline src/Edit.js:20]: dont use snake_case

File: src/Edit.js [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+      };
+      const response = await fetch(`http://localhost:3000/interviews/` + id, options);
```
Comment (joydeep-ib) [inline src/Edit.js:22]: Don't use string concat

File: src/Edit.js [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+      console.log(response);
+      window.location = "/";
```
Comment (joydeep-ib) [inline src/Edit.js:24]: use history push state here

File: src/Edit.js [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+
+  }
+
+
+  console.log("I am in EditInterview ");
```
Comment (joydeep-ib) [inline src/Edit.js:29]: Yaah

File: src/Edit.js [hunk 14]
```diff
@@ -0,0 +1,115 @@
...
+      };
+      const response = await fetch(`http://localhost:3000/interviews/` + id, options);
+      console.log(response);
+      window.location = "/";
+
+  }
+
+
+  console.log("I am in EditInterview ");  
+  event.preventDefault();
```
Comment (joydeep-ib) [inline src/Edit.js:30]: Why do you need preventDefault?

File: src/Edit.js [hunk 15]
```diff
@@ -0,0 +1,115 @@
...
+  const data = new FormData(event.target);
```
Comment (joydeep-ib) [inline src/Edit.js:31]: Why do you need to construct FormData?

File: src/Edit.js [hunk 16]
```diff
@@ -0,0 +1,115 @@
...
+  var json_object   = {}
+  data.forEach((value, key) => {json_object[key] = value});
```
Comment (joydeep-ib) [inline src/Edit.js:33]: Can be improved

File: src/Edit.js [hunk 17]
```diff
@@ -0,0 +1,115 @@
...
+
+  perform_function(json_object);
```
Comment (joydeep-ib) [inline src/Edit.js:35]: Not a good name

File: src/Edit.js [hunk 18]
```diff
@@ -0,0 +1,115 @@
...
+  
+  
+}
+
+export default class Edit extends React.Component{
```
Comment (joydeep-ib) [inline src/Edit.js:40]: Not a good name. Please use a different one

File: src/Edit.js [hunk 19]
```diff
@@ -0,0 +1,115 @@
...
+  state = {
```
Comment (joydeep-ib) [inline src/Edit.js:41]: No class-based components

File: src/Edit.js [hunk 20]
```diff
@@ -0,0 +1,115 @@
...
+    loading: true,
+    interviews: null,
+  };
+
+  async componentDidMount() {
```
Comment (joydeep-ib) [inline src/Edit.js:46]: use useEffect hook

File: src/Edit.js [hunk 21]
```diff
@@ -0,0 +1,115 @@
...
+    
+    // fetching id from current page url
+    const pageUrl = window.location.href
+    const id = pageUrl.substring(35,38);
```
Comment (joydeep-ib) [inline src/Edit.js:50]: See above comments

File: src/Edit.js [hunk 22]
```diff
@@ -0,0 +1,115 @@
...
+    
+    const url = "http://localhost:3000/interviews/" + id;
+    const response = await fetch(url);
+    const data = await response.json();
```
Comment (joydeep-ib) [inline src/Edit.js:54]: Please handle API error responses too

File: src/Edit.js [hunk 23]
```diff
@@ -0,0 +1,115 @@
...
+
+    if(!this.state.interviews ) {
+      return <div>Retrieving Interview data...</div>
+    }
+    
+    let interview = this.state.interviews
+
+      return (
+            <div>
+              <form onSubmit = { EditInterview } className="section">
```
Comment (joydeep-ib) [inline src/Edit.js:73]: Not recommended

File: src/Edit.js [hunk 24]
```diff
@@ -0,0 +1,115 @@
...
+                      </div>
+                      <div className="form-group">
+                        <label>Start Time:</label>
+                        <input type="datetime-local" name="start_time" id="stime"></input>
+                      </div>
+                      <div className="form-group">
+                        <label>End Time:</label>
+                        <input type="datetime-local" name="end_time" id="etime"></input>
+                      </div>
+                      <br></br><br></br>
```
Comment (joydeep-ib) [inline src/Edit.js:97]: No br tags please. Its against the rules of semantic HTML

File: src/Error404.js [hunk 25]
```diff
@@ -0,0 +1,19 @@
+import React from 'react'
+import './App.css'
+
+function Error404() {
+    return (
+      <div>
+        <body class="rails-default-error-page">
```
Comment (joydeep-ib) [inline src/Error404.js:7]: Why are you using body tag inside div tag

File: src/Home.js [hunk 26]
```diff
@@ -0,0 +1,13 @@
+import React from 'react'
+import './App.css'
+
+function Home() {
+    return (
+      <div>
+        <section className="section">
+            <h1>Hello, HR Manager!</h1>
```
Comment (joydeep-ib) [inline src/Home.js:8]: Hi

File: src/Interviews.js [hunk 27]
```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import './App.css';
+import { Link } from  'react-router-dom';
+
+
+export default class Interviews extends React.Component{
```
Comment (joydeep-ib) [inline src/Interviews.js:6]: No classbased components

File: src/Interviews.js [hunk 28]
```diff
@@ -0,0 +1,67 @@
...
+import { Link } from  'react-router-dom';
+
+
+export default class Interviews extends React.Component{
+  state = {
+    loading: true,
+    interviews: [],
+  };
+
+  async componentDidMount() {
```
Comment (joydeep-ib) [inline src/Interviews.js:12]: Same as above

File: src/Bottombar.js (hunk 1 above)
Comment (Sanjit-Prasad) [reply src/Bottombar.js:12]: haha :)

File: src/Bottombar.js (hunk 2 above)
Comment (Sanjit-Prasad) [reply src/Bottombar.js:16]: done

File: src/Delete.js (hunk 3 above)
Comment (Sanjit-Prasad) [reply src/Delete.js:7]: done

File: src/Delete.js (hunk 4 above)
Comment (Sanjit-Prasad) [reply src/Delete.js:16]: done

File: src/App.js (hunk 5 above)
Comment (Sanjit-Prasad) [reply src/App.js:2]: done

File: src/Delete.js (hunk 6 above)
Comment (Sanjit-Prasad) [reply src/Delete.js:3]: changed to DeleteEvent

File: src/Edit.js (hunk 7 above)
Comment (Sanjit-Prasad) [reply src/Edit.js:5]: changed to updateEvent

File: src/Delete.js (hunk 8 above)
Comment (Sanjit-Prasad) [reply src/Delete.js:5]: This worked:         const id = this.props.match.params.id;

File: src/Edit.js (hunk 9 above)
Comment (Sanjit-Prasad) [reply src/Edit.js:8]: ok

File: src/Edit.js (hunk 10 above)
Comment (Sanjit-Prasad) [reply src/Edit.js:20]: done

File: src/Edit.js (hunk 11 above)
Comment (Sanjit-Prasad) [reply src/Edit.js:22]: okay

Comment (Sanjit-Prasad) [reply src/Edit.js:22]: used this: ${id}

File: src/Edit.js (hunk 12 above)
Comment (Sanjit-Prasad) [reply src/Edit.js:24]: okay

File: src/Edit.js (hunk 14 above)
Comment (Sanjit-Prasad) [reply src/Edit.js:30]: not needed.

File: src/Edit.js (hunk 17 above)
Comment (Sanjit-Prasad) [reply src/Edit.js:35]: changed to actionEdit.

File: src/Edit.js (hunk 18 above)
Comment (Sanjit-Prasad) [reply src/Edit.js:40]: made EditEvent

File: src/Edit.js (hunk 21 above)
Comment (Sanjit-Prasad) [reply src/Edit.js:50]: done.

File: src/Edit.js (hunk 22 above)
Comment (Sanjit-Prasad) [reply src/Edit.js:54]: okay

File: src/Error404.js (hunk 25 above)
Comment (Sanjit-Prasad) [reply src/Error404.js:7]: removed

File: src/Edit.js (hunk 24 above)
Comment (Sanjit-Prasad) [reply src/Edit.js:97]: {'\n'} not working, need alternatives.

File: src/Edit.js (hunk 23 above)
Comment (Sanjit-Prasad) [reply src/Edit.js:73]: **what could be better approach then?**

File: src/Bottombar.js (hunk 2 above)
Comment (joydeep-ib) [reply src/Bottombar.js:16]: Where?

File: src/Edit.js (hunk 15 above)
Comment (Sanjit-Prasad) [reply src/Edit.js:31]: not needed.

File: src/Edit.js (hunk 16 above)
Comment (Sanjit-Prasad) [reply src/Edit.js:33]: can't find alternative, use from here
https://stackoverflow.com/questions/41431322/how-to-convert-formdatahtml5-object-to-json/46774073

File: src/Edit.js (hunk 9 above)
Comment (Sanjit-Prasad) [reply src/Edit.js:8]: const id = this.props.match.params.id;
on using this, i am getting the id but data get pushed to url also.

**Pull request merged**